include/zephyr/logging/log_backend.h

```diff
@@ -51,6 +81,12 @@ struct log_backend_api {
 	void (*init)(const struct log_backend *const backend);
 	int (*format_set)(const struct log_backend *const backend,
 				uint32_t log_type);
+
+#ifdef CONFIG_LOG_BACKEND_EVENT_NOTIFICATION_SUPPORT
```

Comment:
i would remove this ifdef, it's only 4(8) bytes in ROM and allows to remove all ifdefs from `log_backend_notify`

Reply:
Removed `#ifdef`.